slf4j-simple/src/main/java/org/slf4j/impl/SimpleLogger.java

```diff
@@ -209,8 +209,15 @@ public static SimpleLogListenerImpl makeLogListener(Hashtable config, PrintStrea
         return new SimpleLogListenerImpl(stream, stackTracePrinter, timeAdjuster, SHOW_DATE_TIME, DATE_FORMATTER, START_TIME, SHOW_THREAD_NAME, LEVEL_IN_BRACKETS, WARN_LEVEL_STRING);
     }
 
+    public static void updateConfig(Hashtable config) {
+        synchronized(SIMPLE_LOGGER_PROPS) {
```

Comment:
While at it, could declare that non-private `SIMPLE_LOGGER_PROPS` field as `final`, so that instance couldn't be replaced between invocations of this update method.

Reply:
Good point!

Reply:
Made it `final`